start.sh

```
@@ -930,25 +1005,156 @@ stop_services() {
+            all_stopped=true
+
+            # Check if any service still has processes running OR ports occupied
+            local i=0
+            for service in "${services[@]}"; do
+                local pid_file="$PID_DIR/${service}.pid"
+                local port="${service_ports[$i]}"
+                local service_stopped=true
+
+                # Check if main process still exists
+                if [ -f "$pid_file" ]; then
+                    local pid=$(cat "$pid_file")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)